website/docs/commands/cli-config.html.markdown

```diff
@@ -63,35 +63,89 @@ The following settings can be set in the CLI configuration file:
-    organization tokens cannot be used for command-line Terraform actions.
-
-    -> **Note:** The credentials hostname must match the hostname in your module
-    sources and/or backend configuration. If your Terraform Enterprise instance
-    is available at multiple hostnames, use one of them consistently. (The SaaS
-    version of Terraform Cloud responds to API calls at both its current
-    hostname, app.terraform.io, and its historical hostname,
-    atlas.hashicorp.com.)
+- `credentials` - configures credentials for use with Terraform Cloud, Terraform
+Enterprise, or other hosts that provide Terraform services. See
+[Credentials](#credentials)
+below for more information.
+
+- `credentials_helper` - configures an external helper program for the storage
+and retrieval of credentials for Terraform Cloud, Terraform Enterprise,
+or other hosts that provide Terraform Services. See
+[Credentials Helpers](#credentials-helpers)
+below for more information.
+
+## Credentials
+
+[Terraform Cloud](/docs/cloud/) provides a number of remote network services
+for use with Terraform, and [Terraform Enterprise](/docs/enterprise/) allows
+hosting those services inside your own infrastructure.
+For example, these systems offer both
+[remote operations](/docs/cloud/run/cli.html) and a
+[private module registry](/docs/cloud/registry/).
+
+When interacting with Terraform-specific network services, Terraform expects
+to find API tokens in CLI configuration files in `credentials` blocks:
+
+```hcl
+credentials "app.terraform.io" {
+  token = "xxxxxx.atlasv1.zzzzzzzzzzzzz"
+}
+```
+
+You can have multiple `credentials` blocks if you regularly use services from
+multiple hosts. Many users will configure only one, for either
+Terraform Cloud (at `app.terraform.io`) or for their organization's own
+Terraform Enterprise host. Each `credentials` block contains a `token` argument
+giving the API token to use for that host.
+
+~> **Important:** If you are using Terraform Cloud or Terraform Enterprise,
+the token provided must be either a
+[user token](/docs/cloud/users-teams-organizations/users.html#api-tokens)
+or a
+[team token](/docs/cloud/users-teams-organizations/api-tokens.html#team-api-tokens);
+organization tokens cannot be used for command-line Terraform actions.
+
+-> **Note:** The credentials hostname must match the hostname in your module
+sources and/or backend configuration. If your Terraform Enterprise instance
+is available at multiple hostnames, use only one of them consistently.
+Terraform Cloud responds to API calls at both its current hostname
+`app.terraform.io`, and its historical hostname `atlas.hashicorp.com`.
+
+If you are running the Terraform CLI interactively on a computer that is capable
+of also running a web browser, you can optionally obtain credentials and save
+them in the CLI configuration automatically using
+[the `terraform login` command](./login.html).
+
+### Credentials Helpers
```

Comment:
What do you think of making this a link to the credentials helper documentation instead of rehashing it here? The commands documentation generally focuses on the most common use-cases and links to alternatives.

Reply:
It makes sense to me to have a short description of how to _use_ a credentials helper in the CLI configuration files. The internals docs are focused on implementing a credentials helper, rather than configuring one for use. Having just a little more detail on usage here seems reasonable to me, and promotes discovery of the feature without distracting.

But I don't feel very strongly about this (★★☆☆☆) and I'd be happy to e.g. move all but the first paragraph over if that's not convincing!

Reply:
Good point! Thanks!